chall11/jbaringo.c

```diff
@@ -0,0 +1,137 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+int check_num(int j, int i, const char *code)
+{
+    if (i > 0 && code[i -1] != '\n')
+        return (0);
+    while (isdigit(code[j]))
+    {
+        if (code[j] != code[i])
+            return (0);
+        j++;
+        i++;
+    }
+    return (i);
+}
+
+char *ft_strjoin(char *s1, char c)
+{
+    char* result;
+    int i = 0;
+
+    if(!(result = malloc(sizeof(char) * strlen(s1) + 1 + 1)))
+        return (NULL);
+    while (s1[i])
+    {
+        result[i] = s1[i];
+        i++;
+    }
+    result[i] = c;
+    result[i+1] = '\0';
+    return (result);
+}
+
+int *goto_reminder(int j, int *loop)
+{
+    int i = 0;
+
+    while (loop[i])
+    {
+        if (loop[i] == j)
+            return (NULL);
+        i++;
+    }
+    int *loop2;
+
+    if (!(loop2 = malloc (sizeof(int) * i + 1)))
+            return (NULL);
+    i = 0;
+    while (loop[i])
+    {
+        loop2[i] = loop[i];
+        i++;
+    }
+    loop2[i] = j;

```

Comment:
You have a `heap-buffer-overflow` here in every test. I presume you go out of bounds at the end of your loop. You should check out `fsanitize` option for `gcc` 👍 . This will sadly make you fail all tests.